flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java

```diff
@@ -833,10 +832,44 @@ public TableResultInternal executeInternal(List<ModifyOperation> operations) {
         return executeInternal(transformations, sinkIdentifierNames, jobStatusHookList);
     }
 
-    private ModifyOperation getOperation(ReplaceTableAsOperation rtasOperation) {
-        // rtas drop table first, then create
+    private ModifyOperation getModifyOperation(
+            ReplaceTableAsOperation rtasOperation, List<JobStatusHook> jobStatusHookList) {
         CreateTableOperation createTableOperation = rtasOperation.getCreateTableOperation();
         ObjectIdentifier tableIdentifier = createTableOperation.getTableIdentifier();
+        Catalog catalog = catalogManager.getCatalog(tableIdentifier.getCatalogName()).orElse(null);
+        ResolvedCatalogTable catalogTable =
+                catalogManager.resolveCatalogTable(createTableOperation.getCatalogTable());
+        Optional<DynamicTableSink> stagingDynamicTableSink =
+                getSupportsStagingDynamicTableSink(createTableOperation, catalog, catalogTable);
+        if (stagingDynamicTableSink.isPresent()) {
+            // use atomic rtas
+            DynamicTableSink dynamicTableSink = stagingDynamicTableSink.get();
+            if (!rtasOperation.isCreateOrReplace()) {
```

Comment:
I'm wondering what's the behavior for staging replace in spark.
Is it also will check the table exists or not? And if not, will it also throw an exception?

Reply:
I checked the spark code and it will check the table exists or not, if not will throw an exception.
![image](https://github.com/apache/flink/assets/9247964/df1d3c86-ab2b-4ab5-b46b-3b103b1458d4)